sensor/common/registry/registry_store.go

```diff
@@ -490,28 +503,28 @@ func (rs *Store) upsertSecretByName(namespace, secretName string, dockerConfig c
 		}
 
 		if isGlobalPullSecret {
-			if err := rs.upsertGlobalRegistry(registryAddr, dce); err != nil {
-				log.Errorf("Upserting global registry for pull secret %q, namespace %q, address %q: %v", secretName, namespace, registryAddr, err)
+			if err := rs.upsertGlobalRegistry(registryAddr, host, dce); err != nil {
+				log.Errorf("Unable to upsert global registry for pull secret %q, namespace %q, registry %q, host %q: %v", secretName, namespace, registryAddr, host, err)
 			}
 		}
 
 		// Regardless if this secret is the global pull secret, we still store it
 		// in case there is a workload that directly references it by name.
-		rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
+		rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, host, dce)
 	}
 
 	log.Debugf("Upserted %d entries from secret %q in namespace %q", len(dockerConfig), secretName, namespace)
 }
 
-func (rs *Store) upsertPullSecretByNameNoLock(namespace, secretName, registryAddr string, dce config.DockerConfigEntry) {
-	registryAddr = urlfmt.TrimHTTPPrefixes(registryAddr)
+func (rs *Store) upsertPullSecretByNameNoLock(namespace, secretName, registry, host string, dce config.DockerConfigEntry) {
+	host = urlfmt.TrimHTTPPrefixes(host)
```

Comment:
Apologies for the confusion from my last comment - the `host` shouldn't need trimming (will never have an `http` prefix), it was the 'registry' that could have the prefix - however given your latest changes, specifically the parsing in `isRegistryMatch`, prefixes are handled nicely.

As a result trimming isn't necessary here anymore (feel free to remove). Apologies again for the confusion.